lib/widgets/compose_box.dart

```diff
@@ -953,24 +953,18 @@ class _FixedDestinationComposeBoxState extends State<_FixedDestinationComposeBox
 }
 
 class ComposeBox extends StatelessWidget {
-  const ComposeBox({super.key, this.controllerKey, required this.narrow});
+  const ComposeBox({super.key, this.controllerKey, required Narrow narrow}) : _narrow = narrow;
 
+  final Narrow _narrow;
   final GlobalKey<ComposeBoxController>? controllerKey;
-  final Narrow narrow;
 
   @override
   Widget build(BuildContext context) {
-    final narrow = this.narrow;
-    if (narrow is StreamNarrow) {
-      return _StreamComposeBox(key: controllerKey, narrow: narrow);
-    } else if (narrow is TopicNarrow) {
-      return _FixedDestinationComposeBox(key: controllerKey, narrow: narrow);
-    } else if (narrow is DmNarrow) {
-      return _FixedDestinationComposeBox(key: controllerKey, narrow: narrow);
-    } else if (narrow is CombinedFeedNarrow) {
-      return const SizedBox.shrink();
-    } else {
-      throw Exception("impossible narrow"); // TODO(dart-3): show this statically
-    }
+    return switch (_narrow) {
+      StreamNarrow() => _StreamComposeBox(key: controllerKey, narrow: _narrow),
+      TopicNarrow() => _FixedDestinationComposeBox(key: controllerKey, narrow: _narrow),
+      DmNarrow() => _FixedDestinationComposeBox(key: controllerKey, narrow: _narrow),
```

Comment:
I think this one will be a bit easier to read as a switch statement (rather than expression).

That way the returned expressions remain aligned with each other, and also less indented — these lines get long.